core/src/main/java/com/netflix/iceberg/GenericDataFile.java

```diff
@@ -405,4 +406,17 @@ public String toString() {
         .toString();
   }
 
+  private static <K, V> Map<K, V> copy(Map<K, V> map) {
+    if (map != null) {
+      return ImmutableMap.copyOf(map);
```

Comment:
This will throw a NPE if the `map` parameter is a `HashMap` with null keys or values. Is that expected?
Would this be considered overly defensive programming?
```
        map.values().removeIf(Objects::isNull);
        map.keySet().removeIf(Objects::isNull);
```

Reply:
There should be no null keys in these maps because they are built internally. If these were passed from user code, I would agree.